addons/website_slides/static/src/js/slides_slide_preview.js

```diff
@@ -0,0 +1,90 @@
+odoo.define('website_slides.slide.preview', function (require) {
+    'use strict';
+
+    var publicWidget = require('web.public.widget');
+    var Dialog = require('web.Dialog');
+    var core = require('web.core');
+    var _t = core._t;
+
+    var SlidePreviewDialog = Dialog.extend({
+        template: 'slides.slide.preview',
+
+        /**
+         * @override
+         */
+        init: function (parent, options) {
+            options = _.defaults(options || {}, {
+                title: _t('Preview Slide'),
+                size: 'medium',
+                buttons: [{
+                    text: _t('Yes'),
+                    classes: 'btn-primary',
+                    click: this._onClickPreview.bind(this)
+                }, {
+                    text: _t('No'),
+                    close: true
+                }]
+            });
+
+            this.$slideTarget = options.slideTarget;
+            if (this.$slideTarget.hasClass('badge-success')) {
+                this.action = _t('deactivate');
```

Comment:
I think it would be better to have two strings in the qweb template, depending on the widget's state. Indeed it eases translations to have two complete strings to translate instead of a single one with an inclusion that external people will not easily find. When translating strings people have no easy access to the context in which it is used.